### PR TITLE
AWS CIS v3.0.0 - Section 3 - Logging

### PR DESCRIPTION
**Pull request by vkumbha**

### CODE REVIEW

**Comment by dboeke** [comment]
@vkumbha per the standards doc:  https://turbothq.slack.com/archives/C06DF1TE16D/p1715141992661679?thread_ts=1715088393.985959&cid=C06DF1TE16D   
```
multi-line strings should use the [indented heredoc](https://developer.hashicorp.com/terraform/language/expressions/strings#indented-heredocs) `<<-EOT` style.
```

File: cis_enforcements/aws/cis_v300_s3_logging/bucket.tf [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  value    = "Check: Enabled to Access Logging > Bucket"
+  # value = "Enforce: Enabled to Access Logging > Bucket"
+}
+
+# AWS > S3 > Bucket > Access Logging > Bucket
+resource "turbot_policy_setting" "aws_s3_bucket_access_logging_bucket" {
+  resource       = turbot_smart_folder.aws_cis_v300_s3_logging.id
+  type           = "tmod:@turbot/aws-s3#/policy/types/bucketAccessLoggingBucket"
+  note           = "AWS CIS v3.0.0 - Controls: 3.04"
+  template_input = <<EOT
```
Comment (dboeke) [inline cis_enforcements/aws/cis_v300_s3_logging/bucket.tf:15]: let's use `<<-EOT` for all multiline strings and not `EOV` etc..

File: cis_enforcements/aws/cis_v300_s3_logging/common.tf [hunk 2]
```diff
@@ -0,0 +1,12 @@
+resource "turbot_smart_folder" "aws_cis_v300_s3_logging" {
```
Comment (dboeke) [inline cis_enforcements/aws/cis_v300_s3_logging/common.tf:1]: add `# Smart Folder` comment

File: cis_enforcements/aws/cis_v300_s3_logging/main.tf [hunk 3]
```diff
@@ -0,0 +1,16 @@
+terraform {
+  required_providers {
+    turbot = {
+      source = "turbot/turbot"
+    }
+  }
+}
+
+provider "turbot" {
```
Comment (dboeke) [inline cis_enforcements/aws/cis_v300_s3_logging/main.tf:9]: get rid of the profile variable, per the doc and the example.

File: cis_enforcements/aws/cis_v300_s3_logging/trail.tf [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+  note     = "AWS CIS v3.0.0 - Controls: 3.05"
+  value    = "Check: Encryption at Rest > Customer Managed Key"
+  # value = "Enforce: Encryption at Rest > Customer Managed Key"
+}
+
+# AWS > CloudTrail > Trail > Encryption at Rest > Customer Managed Key
+resource "turbot_policy_setting" "aws_cloudtrail_trail_encryption_at_rest_customer_managed_key" {
+  resource = turbot_smart_folder.aws_cis_v300_s3_logging.id
+  type     = "tmod:@turbot/aws-cloudtrail#/policy/types/trailEncryptionAtRestCustomerManagedKey"
+  note     = <<EON
```
Comment (dboeke) [inline cis_enforcements/aws/cis_v300_s3_logging/trail.tf:93]: note fields don't support Multi-line strings.  keep the note simple and add comments above the policy setting if needed.

**Review by dboeke — CHANGES_REQUESTED** [review]

File: cis_enforcements/aws/cis_v300_s3_logging/main.tf [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+}
+
+variable "logging_bucket" {
+  type        = string
+  description = "The name of an S3 bucket to which the Guardrails Trail will be delivered. If not provided, Turbot will create one."
+  default     = ""
+}
+
+variable "encryption_key" {
```
[the remark this answers is not in the document]
Comment (vkumbha) [reply cis_enforcements/aws/cis_v300_s3_logging/main.tf]: Log file SSE-KMS encryption is set to Enabled by default. CIS 3.05 is looking for CMK located in the same region as the S3 bucket with necessary a KMS Key policies.

**Review by SumitPopat — APPROVED** [review]

**Pull request merged**